### ISSUE TITLE
Fix S4070: False positive

### ISSUE THREAD

**Issue by Chelaris182** [SonarSource/sonar-csharp#809]
The current implementation of the rule raises violation of S4070 for the following code:
```
    [Flags]
    public enum Foo
    {
        N1 = 1,
        N2 = 2,
        N3 = N1 | N2,
        N4 = 4,
        N5 = N4 | N1
    }
```

If you remove N5, the violation goes away.

**Comment by michalb-sonar** [comment]
Thanks for report and sample code @Chelaris182. I was able to reproduce the problem, we'll fix that.

**Comment by Chelaris182** [comment]
Looking at the rule code I also came up with a false negative case:
```
    [Flags]
    public enum Options
    {
        N1 = 1,
        N2 = 2,
        N3 = N1 | N2,
        N5 = 5
    }
```

**Comment by michalb-sonar** [comment]
Good catch @Chelaris182!
